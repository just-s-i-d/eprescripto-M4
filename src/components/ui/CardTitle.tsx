type CardTitlePropsType = {
  children: React.ReactNode;
  className?: string;
};
const CardTitle = ({ children, className }: CardTitlePropsType) => {
  return <h1 className={`text-xl font-bold mb-6 ${className}`}>{children}</h1>;
};
export default CardTitle;
