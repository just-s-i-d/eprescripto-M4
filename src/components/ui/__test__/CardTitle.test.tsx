import { render } from "@testing-library/react";
import "@testing-library/jest-dom";
import CardTitle from "../CardTitle";

describe("Card Title Test", () => {
  const title = "Test Title";
  const className = "test";

  it("matches snapshot", () => {
    const { asFragment } = render(
      <CardTitle className={className}>{title}</CardTitle>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("Title test", () => {
    const { getByText } = render(
      <CardTitle className={className}>{title}</CardTitle>,
    );
    const titleElement = getByText(title);
    expect(titleElement).toBeInTheDocument();
    expect(titleElement).toHaveClass(className);
  });
});
