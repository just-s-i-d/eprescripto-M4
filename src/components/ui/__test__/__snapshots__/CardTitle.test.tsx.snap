// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Title Test matches snapshot 1`] = `
<DocumentFragment>
  <h1
    class="text-xl font-bold mb-6 test"
  >
    Test Title
  </h1>
</DocumentFragment>
`;
